### PR TITLE
python3Packages.pygments: 2.13.0 -> 2.14.0

### PR DESCRIPTION
**Pull request by ncfavier**
https://github.com/pygments/pygments/releases/tag/2.14.0

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ncfavier** [comment]
Anything else I should test beyond basic functionality of pygments itself?

Ofborg seems useless here...

**Comment by SuperSandro2000** [comment]
@ofbrog eval

**Comment by SuperSandro2000** [comment]
> Anything else I should test beyond basic functionality of pygments itself?

We are running tests, so should be fine.

ofborg fails because cache is missing.

**Comment by ncfavier** [comment]
No dice. `python3Packages.pygments.tests` builds fine on the PR's tip (not rebased on staging).

**Comment by SuperSandro2000** [comment]
It doesn't build at all on rebased on staging because sphinx tests fail right now.

**Comment by ncfavier** [comment]
Where do we go from here? Surely I can't be expected to rebuild the world on my laptop just to update a package that runs fine on master?

**Comment by SuperSandro2000** [comment]
We wait until that is fixed :P

@ofborg eval

**Comment by SuperSandro2000** [comment]
I think the failure is caused by the pygments update. I couldn't produce it before.

**Comment by ncfavier** [comment]
https://github.com/sphinx-doc/sphinx/commit/965768bfda2a00ba6466cdb12a7a46efdce47023

Why is our sphinx so outdated?

**Comment by ncfavier** [comment]
https://github.com/NixOS/nixpkgs/pull/210338

**Comment by ncfavier** [comment]
Since the sphinx update might take time, I've just disabled the failing tests.

**Comment by ncfavier** [comment]
Is this good enough to merge?

File: pkgs/development/python-modules/sphinx/default.nix [hunk 1]
```diff
@@ -106,6 +106,12 @@ buildPythonPackage rec {
     # requires imagemagick (increases build closure size), doesn't
     # test anything substantial
     "test_ext_imgconverter"
+
+    # fails with pygments 2.14
+    # TODO remove for sphinx 6
+    "test_viewcode"
+    "test_additional_targets_should_be_translated"
+    "test_additional_targets_should_not_be_translated"
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/sphinx/default.nix:114]: Does this break the actual code? I am a bit reluctant.

Comment (ncfavier) [reply pkgs/development/python-modules/sphinx/default.nix:114]: Judging by https://github.com/sphinx-doc/sphinx/commit/965768bfda2a00ba6466cdb12a7a46efdce47023 it seems like minor formatting differences.

**Review by SuperSandro2000 — APPROVED** [review]
@ofborg build python310Packages.pygments python310Packages.sphinx

**Comment by SuperSandro2000** [comment]
I just pushed an update for rich to this branch because the latest update had a fix for pygments.

**Comment by ncfavier** [comment]
The rich update introduces a circular dependency (for python 3.11 only):

```
rich → markdown-it-py
     → pytest-regressions (tests)
     → pandas
     → hypothesis (tests)
     → sphinx-codeautolink (doc)
     → ipython
     → stack-data
     → executing
     → rich (tests, python >= 3.11)
```

**Comment by ncfavier** [comment]
Broke the cycle at `hypothesis` since there seems to be precedent for this.

**Comment by ncfavier** [comment]
rich 13.3.0 breaks textual...

**Review by joelkoen — APPROVED** [review]

**Comment by ncfavier** [comment]
This is still broken: now ipython's tests are failing.

**Comment by mweinelt** [comment]
Step by step.

**Comment by ncfavier** [comment]
Okay, https://github.com/NixOS/nixpkgs/pull/212441 should fix ipython via https://github.com/ipython/ipython/pull/13882.

File: pkgs/development/python-modules/textual/default.nix [hunk 2]
```diff
@@ -48,6 +48,8 @@ buildPythonPackage rec {
     typing-extensions
   ];
 
+  doCheck = false; # broken with rich 13.3.0
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/textual/default.nix:51]: Please also add why

File: pkgs/development/python-modules/pandas/default.nix [hunk 3]
```diff
@@ -49,7 +49,8 @@ buildPythonPackage rec {
 
   nativeCheckInputs = [
     glibcLocales
-    hypothesis
+    # hypothesis indirectly depends on pandas to build its documentation
+    (hypothesis.override { enableDocumentation = false; })
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pandas/default.nix:53]: Not sure if we should move this to python-packages

File: pkgs/development/python-modules/textual/default.nix (hunk 2 above)
Comment (ncfavier) [reply pkgs/development/python-modules/textual/default.nix]: I've updated textual instead (which required a markdown-it-py update), now tests pass.

File: pkgs/development/python-modules/pandas/default.nix (hunk 3 above)
Comment (ncfavier) [reply pkgs/development/python-modules/pandas/default.nix:53]: Presumably we still want the top-level `hypothesis` to have documentation?

**Comment by ncfavier** [comment]
`pygments`, `pygments.tests` build fine on master with https://github.com/NixOS/nixpkgs/pull/215904 applied. Maybe it's time to merge both PRs?

File: pkgs/development/python-modules/pandas/default.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/pandas/default.nix:53]: I mean if we should move the overwrite to the attr set passed to pandas callPackage.

**Comment by SuperSandro2000** [comment]
It is probably a better idea to wait for a python-updates run to fix possible other regressions in big style.

**Comment by mweinelt** [comment]
Meh. wrong destination branch hit.

**Pull request merged**